mock/api.js

```diff
@@ -297,8 +284,15 @@ export const mockSession = {
 export default {
   'GET /controllers/months': mockIndices,
   'POST /controllers/list': mockControllers,
-  'POST /datasets/list': mockResults,
-  'POST /datasets/detail': mockDetail,
+  'POST /datasets/list': (req, res) => {
+    const data = {};
+    data[req.body.controller] = mockResults(req.body.controller);
+    res.send(data);
```

Comment:
as we are reassigning to a const, we can do something like let data = {} or return {req.body.controller: mockResults(req.body.controller)}

Reply:
@aquibbaig I believe we can't send the response object like that. It's throwing error.
<img width="522" alt="Screenshot 2021-07-01 at 6 02 05 PM" src="https://user-images.githubusercontent.com/16955978/124124816-866a6000-da96-11eb-8345-5ab92ed25f84.png">

Reply:
I have changed the `const` to `let` though

Reply:
@aquibbaig let me know if you have any suggestions for further modification here.

Reply:
The `const` is fine because the JSON document itself is only created once; what's weird about the `= {req.body.controller: ...}` form is that you're trying to assign multiple levels of JSON at once, which doesn't work. It'd have to be `{req: {body: {controller: ...}}}` instead. Which may or may not be easier to read.

However, I think that what you're actually doing here is assigning the literal key `"controller"` in your JSON, whereas the real value is going to be the name of the specific controller.